doc/compiler/optimizer/LoopOptimizationSummary.md

```diff
@@ -0,0 +1,114 @@
+also eliminates bound checks where the range of values of the index inside
+the loop is discovered (need to be a compile time constant) and checks are
+inserted outside the loop so that no bound checks fail inside the loop.
+
+This analysis uses induction variable information found by [Value Propagation](https://github.com/eclipse/omr/blob/master/doc/compiler/optimizer/ValuePropagation.md).
+Another versioning test that is emitted ensures the loop will not run for a
+very long period of time (in which case the async check inside the loop can
+be removed). This special versioning test depends on the number of trees
+(code) inside the loop.
+
+Loop Specialization replaces loop-invariant expressions that are profiled
+and found to be constants, with the constant value after inserting a test
+outside the loop that compares the value to the constant. Note that this
+cannot be done in the the absence of value profiling infrastructure.
+
+## Loop Unrolling and Peeling
+A majority of loops can be unrolled or peeled with or without the loop driving
+tests done after each iteration (loop) body. Usually a peel is inserted before
+the unrolled loop and the residual iterations to be done (at most u-1 residual
+iterations where u is the unroll factor) are also done using the peeled code.
+Peeling aids [Partial Redundancy Elimination (PRE)](optimizer/GlobalOptimizationsSummary.md)
+as code does not have to be moved; instead commoning of dominated expressions can be done
+(which is far easier than code motion because of problems introduced by exception checks
+in Java).
+
+Async checks (yield points) are eliminated from u-1 unrolled bodies and only
+one remains. Unroll factors and code growth thresholds are arrived at based
+on profiling information when available. This analysis uses induction variable
+information found by [Value Propagation](https://github.com/eclipse/omr/blob/master/doc/compiler/optimizer/ValuePropagation.md).
+
+## Loop Invariant Code Motion
+Done as part of [Partial Redundancy Elimination (PRE)](optimizer/GlobalOptimizationsSummary.md)
+
+## Loop Inversion
+Loop Inversion converts a loop where the induction variable is counting up
+from zero into one where it counts down to zero. Note that this can only be
+done if the inversion of the loop does not affect program semantics inside
+the loop (order of exceptions thrown etc). The benefit of doing this is the
+fact that there are, in general, instructions that can perform compare/branch
+against zero in an efficient manner and special counter machine register
+(for e.g. On PowerPC) can be used.
+
+## Loop Striding
+Loop Striding creates derived induction variables (e.g. the address calculation
+for array accesses `&a[i]`) and increment/decrement the derived induction
+variables by the appropriate stride through every iteration of the loop.
+This can eliminate some complex array address calculations inside the loop,
+instead replacing by a load (reg or memory). Note that we also create derived
```

Comment:
```suggestion
This can eliminate some complex array address calculations inside the loop,
replaced instead by a load (reg or memory). Note that we also create derived
```